internal/test/data/vector/gen.go

```diff
@@ -131,3 +131,38 @@ func gaussianDistributedUint8VectorGenerator(n, dim int, mean, sigma float64) []
 		}
 	})
 }
+
+func GenF32Vec(dist Distribution, num int, dim int) ([][]float32, error) {
+	generator, err := Float32VectorGenerator(dist)
+	if err != nil {
+		return nil, err
+	}
+	return generator(num, dim), nil
+}
+
+func GenIntVec(dist Distribution, num int, dim int) ([][]float32, error) {
```

Comment:
This function should be changed the name because it returns uint8 vectors.
```suggestion
func GenUint8Vec(dist Distribution, num int, dim int) ([][]float32, error) {
```
or
```suggestion
func GenU8Vec(dist Distribution, num int, dim int) ([][]float32, error) {
```
U8 is based on usigned 8bit integer in Rust `u8`.